tests/conftest.py

```
@@ -49,3 +50,42 @@ def cleanup_fixture(should_do_global_cleanup_after_test: bool):
 def dynamo_reset():
     yield
     torch._dynamo.reset()
+
+
+@pytest.fixture(scope="function")
+def remote_openai_server(request):
+    """ Fixture to set up a test server."""
+
+    params = request.node.callspec.params
+
+    # Extract parameters from the test function for the server
+    model = params['model']
+    warmup_shape = params['warmup_shape']
+    backend = params['backend']
+    vllm_version = params['vllm_version']
```

In general I liked how the wrapping works here. However, what happens if the user forget to declare this params? Should we throw an exception or try to get a better message to inform the user about this?